autoscale/src/main/java/com/sequenceiq/periscope/controller/AutoScaleClusterCommonService.java

```diff
@@ -114,28 +117,39 @@ public Cluster setStopStartScalingState(Long clusterId, Boolean enableStopStartS
         Cluster cluster = clusterService.findById(clusterId);
         if (enableStopStartScaling != null && !cluster.isStopStartScalingEnabled().equals(enableStopStartScaling)) {
             cluster = clusterService.setStopStartScalingState(cluster, enableStopStartScaling);
-            processAutoscalingStateChanged(cluster);
         }
         return cluster;
     }
 
     public void deleteAlertsForClusterCrn(String stackCrn) {
         Cluster cluster = getClusterByCrnOrName(NameOrCrn.ofCrn(stackCrn));
+        deleteStoppedNodesIfPresent(cluster.getId());
         cluster = clusterService.deleteAlertsForCluster(cluster.getId());
-        processAutoscalingStateChanged(cluster);
+        processAutoscalingConfigChanged(cluster);
     }
 
     public void deleteAlertsForClusterName(String stackName) {
         Cluster cluster = getClusterByCrnOrName(NameOrCrn.ofName(stackName));
+        deleteStoppedNodesIfPresent(cluster.getId());
```

Comment:
Should we make this call after persisting the DB state ? Also in the crn method.

Reply:
Updated this and the CRN method to trigger node deletion after persisting the cluster in the DB